code/js/popup.js

```diff
@@ -14,6 +15,24 @@ function AddNewTags (tagName)
 	tags.push(tagName);
 }
 
+function updateFavs(x, filename) {
```

Comment:
Local storage will be deleted when the user will clear his/ her browsing history with cookies and other site data option selected. 

An ideal user may clear history occasionally but we do not want to lose all the data. 

So, you may explore other options like saving in a file in appropriate format.